pkg/controller/revision/revision.go

```diff
@@ -967,7 +969,7 @@ func (c *Controller) receiveControllerConfig(configMap *corev1.ConfigMap) {
 func (c *Controller) getResolver() resolver {
 	c.resolverMutex.Lock()
 	defer c.resolverMutex.Unlock()
-	return c.resolver
+	return c.resolver.Clone()
```

Comment:
I think this is not needed, as the place where `c.resolver` is set (`receiveControllerConfig()`) and the place where it is read (here) are both protected by the mutex, and the thing returned doesn't reference mutable fields of the owning Controller.

So you have the guarantee that the resolver that is going to be used is consistent at the time you get it. The next step would be to synchronize the whole `createDeployment()` (or the call to `Resolve()` in a `func()`, but according to @glyn this is impractical.

So long story short, I don't think this actually solves a problem.

Reply:
Agreed. Thanks!